### DISCUSSION TITLE
Update connection to use SelectConnection

### DISCUSSION THREAD

**Discussion by marcosstefani** [ElaraDevSolutions/lepus#2] [Ideas]
@lukebakken identifies:
"I strongly suggest using the SelectConnection class and running it in its own thread. Otherwise, more than likely a user will block Pika's I/O loop unintentionally."
"Another issue your library has is that once you start consuming, you will block, so you really can't both publish and consume at the same time."

**Comment by marcosstefani** [comment]
@lukebakken I created the branch https://github.com/ElaraDevSolutions/lepus/tree/6-update-connection-to-use-selectconnection.

Im trying to change. Still no success, but if you want to take a look, feel free.

**Comment by lukebakken** [comment]
You are accessing the connection from two different threads. Remember that Pika is not thread-safe.
